plugins/node.d/nutups_

```diff
@@ -11,22 +11,22 @@
 #%# family=contrib
 #%# capabilities=autoconf suggest
 
-UPS=$(basename $0 | cut -d_ -f2)
-FUNCTION=$(basename $0 | cut -d_ -f3)
-UPSC=$(which upsc)
+UPS=$(basename "$0" | cut -d_ -f2)
+FUNCTION=$(basename "$0" | cut -d_ -f3)
+UPSC=$(command -v upsc)
 
 if [ "$1" = "autoconf" ]; then
 	[ -x "$UPSC" ] && [ -r /etc/nut/ups.conf ] && echo yes && exit 0
-	echo "upsc or /etc/nut/ups.conf not found"
+	echo "no (upsc or /etc/nut/ups.conf not found)"
 	exit 0
 fi
 
 if [ "$1" = "suggest" ]; then
 	grep '^\[[^]]*\]$' /etc/nut/ups.conf \
 		| tr -d '][' \
-		| while read ups; do
+		| while read -r ups; do
 			for i in voltages freq charge current; do
-				echo ${ups}_${i}
+				echo "${ups}"_${i}
```

Comment:
Just a style note: you could have also quoted the full string up to the end - this is usually a bit easier to read.
But this is just a matter of taste ...